<!DOCTYPE html>
<!-- saved from url=(0095)http://www.gparc.org/docs/urn:lsid:8080.gpbroad.broadinstitute.org:genepatternmodules:345:1.4.9 -->
<html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>ConstellationMap (v1.4.x)</title>
  <link href="./application.css" media="all" rel="stylesheet">
  <script src="./application.js"></script><style>.cke{visibility:hidden;}</style><style type="text/css"></style>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta content="authenticity_token" name="csrf-param">
<meta content="ApMfz9igCh9SAwERmUbadoKTMneA+NPCaaZvitL6JFg=" name="csrf-token">
  <link href="http://www.gparc.org/favicon.ico" rel="shortcut icon" type="image/vnd.microsoft.icon">
</head>
<body>

		<div class="fluid module-top">
		  <div class="container section-pad-sm">
		    <div class="row">
		      <div class="col-md-12">	

		<div class="row">
			<div class="col-md-9" style='background-image:url("front-banner.jpg");color:#fff'>
				<h1>ConstellationMap, v1.4.x <a style="float: right" href="https://www.genepattern.org"><img alt="GP Logo" src="gplogo.png" /></a></h1>
			</div>
			<div class="col-md-3">
			</div>
		</div>	
		<p style="font-size: 1.2em; margin-top: 20px;">A downstream visualization and analysis tool for gene set enrichment results with interactive web visualizer, to be run on the GenePattern platform.</p>
		<div class="row">
			<div class="col-sm-4">
				<p><strong>Author: </strong>Felix Wu;Broad Institute</p>
			</div>
			<div class="col-sm-4">
				<p><strong>Contact: </strong>gp-help@broadinstitute.org
			</p></div>
			<div class="col-sm-4">
				<p><strong>Algorithm Version: </strong>1.2
			</p></div>
		</div>
	</div>
</div>

<div class="row">
	<div class="col-md-12">
		<h2>Introduction</h2>

<p style="line-height: 20.8px;">ConstellationMap helps leverage the full power of a gene set enrichment analysis by identifying commonalities between high-scoring gene sets and mapping their relationships. ConstellationMap visualizes in an interactive web application the enrichment profile similarity and gene member overlap of gene sets, which are positively or negatively enriched in relation to a phenotype in a two-class or continuous-class comparison. It uses provided enrichment scores (e.g., those generated by&nbsp;<a href="http://www.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection">ssGSEAProjection</a>) to estimate normalized mutual information (NMI) scores and project top scoring sets onto a circular plot with the following features:</p>

<p style="line-height: 20.8px;">(1) Gene sets are represented as&nbsp;<strong>nodes</strong>&nbsp;with radial distance to the center reflecting positive or negative enrichment in the phenotype, whichever direction is specified. Higher associations for the correlation direction are plotted closer to the center.</p>

<p style="line-height: 20.8px;">(2) Member gene overlaps between sets are represented as&nbsp;<strong>edges</strong>&nbsp;connecting nodes with thickness proportional to degree of overlap.</p>

<p style="line-height: 20.8px;">(3) The angular distance between nodes is relatively proportional to the similarity of their enrichment profiles, i.e. more similar enrichment patterns have a smaller angular distance.</p>

<p style="line-height: 20.8px;"><strong>See the video tutorial</strong>:&nbsp;<a href="http://www.youtube.com/watch_popup?v=puadjX_xh8g">Using the Constellation Map Visualizer</a></p>

<p style="line-height: 20.8px;"><strong>See the&nbsp;<em>F1000Research&nbsp;</em>paper (Tan Y, Wu F, Tamayo P&nbsp;<em>et al.</em>, 2015):&nbsp;</strong>PubMed (<a href="http://www.ncbi.nlm.nih.gov/pubmed/26594333">http://www.ncbi.nlm.nih.gov/pubmed/26594333</a>), DOI (<a href="http://dx.doi.org/10.12688/f1000research.6644.1">10.12688/f1000research.6644.1)</a></p>

<p style="line-height: 20.8px;">Web interactive features allow export of selected overlapping gene symbols for annotation in&nbsp;<a href="http://david.abcc.ncifcrf.gov/">DAVID</a>,&nbsp;<a href="http://www.msigdb.org/">MSigDB</a>, and&nbsp;<a href="http://genemania.org/">GeneMania</a>. Click on the output file&nbsp;<em>Visualizer.html</em>&nbsp;to open the interactive ConstellationMap plot from the Jobs Tab.</p>

<ul style="line-height: 20.8px;">
	<li>Use mouse click on the web visualization to select a node or an edge. Use shift+click, or click-drag to select a group of features.</li>
	<li>Selection displays gene set names, number of genes in the set, number of sets used to compute overlap, and overlapping member gene symbols.</li>
	<li>Adjust number of nodes in highlighted area for which to display the maximum number of overlapping gene symbols.</li>
</ul>

<p style="line-height: 20.8px;">In addition to the web application, ConstellationMap outputs two static plots: (1) a positive red to negative blue heat map of per sample enrichment of the gene sets ranked by NMI scores as well as Area Under the Curve (AUC) and t-test metrics along with corresponding p-values, and (2) a constellation map marking the phenotype of interest in the center in red, concentric contour arcs marking association to the phenotype (as measured by NMI), nodes as numbered open circles, overlap as green lines, and a key of the numbered gene sets.</p>

<h2>Example Workflows</h2>

<p style="line-height: 20.8px;">Users may input gene set enrichment data produced by any algorithm as long as it is in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GCT">GCT format</a>, as well as any gene set collection in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMX">GMX</a>&nbsp;or&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMT">GMT</a>&nbsp;format. For convenience, GenePattern modules including ConstellationMap provide MSigDB gene set collection files from a drop-down menu. The two module workflow examples presented here both use outputs from GenePattern’s&nbsp;<a href="http://www.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection">ssGSEAProjection module</a>&nbsp;as inputs to ConstellationMap. Workflow (A) uses ssGSEAProjection only on user-specified enriched sets, e.g. those previously identified from another analysis, while Workflow (B) uses enrichment rankings computed by ssGSEAProjection on the expression dataset in ConstellationMap. Users may choose Workflow (A) if they have a preferred method (e.g., GSEA) for identifying gene sets highly associated with a phenotypic class. If users do not have a preferred method, then they may use Workflow (B), which uses NMI to identify highly associated gene sets.&nbsp;See the following figure for a visual representation of the two workflows.</p>

<p style="line-height: 20.8px;"><img alt="ConstellationMap Workflow Figure" src="./content_constellationmap_workflow.png" style="width: 500px; height: 436px; border-width: 1px; border-style: solid;"></p>

<h3>Workflow (A)</h3>

<p style="line-height: 20.8px;">Whole Transcriptome Expression Data → GSEA (or other gene set analysis method) → ssGSEAProjection → ConstellationMap</p>

<ol style="line-height: 20.8px;">
	<li>From a gene set enrichment analysis result using an expression dataset and a preferred gene set collection, e.g. a GSEA module analysis of RNA-Seq expression data using MSigDB’s C3 Motifs Collection, identify a group of interesting gene sets. For example, all gene sets that GSEA identified as highly enriched based on an FDR cutoff of 0.25. Organize the names of these gene sets exactly as they appear in the gene set collection into a comma-separated list without spaces</li>
	<li>Run the ssGSEAProjection module with the following parameter settings:
	<ol style="list-style-type: lower-alpha;">
		<li><em>input gct file</em>: expression dataset used in step (1)</li>
		<li><em>gene sets database (file)</em>: the same file or contains the same gene sets used in step (1) in GMX or GMT format</li>
		<li><em>gene set selection</em>: comma-separated list of interesting gene sets</li>
		<li><em>combine mode</em>: combine.off</li>
	</ol>
	</li>
	<li>Run the ConstellationMap module with the following parameter settings:
	<ol style="list-style-type: lower-alpha;">
		<li><em>input gct file</em>: output from ssGSEAProjection in step (2)</li>
		<li><em>input cls file</em>: same as used in the example GSEA analysis of step (1)</li>
		<li><em>gene sets database (file)</em>: same as used in step (2)</li>
		<li><em>top n</em>: set to a number equal to or less than the number of interesting gene sets inputted at step (2c)</li>
	</ol>
	</li>
</ol>

<h3>Workflow (B)</h3>

<p style="line-height: 20.8px;">Whole Transcriptome Expression Data → ssGSEAProjection&nbsp; → ConstellationMap</p>

<ol style="line-height: 20.8px;">
	<li>Run the ssGSEAProjection module with the following parameter settings:
	<ol style="list-style-type: lower-alpha;">
		<li><em>input gct file</em>: expression dataset</li>
		<li><em>gene sets database (file)</em>: select an MSigDB collection from the drop-down menu or provide a GMX or GMT file of gene sets</li>
		<li><em>gene set selection</em>: ALL</li>
		<li><em>combine mode</em>: combine.off</li>
	</ol>
	</li>
	<li>Run the ConstellationMap module with the following parameter settings:
	<ol style="list-style-type: lower-alpha;">
		<li><em>input gct file</em>: output from ssGSEAProjection in step (1)</li>
		<li><em>input cls file</em>: CLS file containing two categorical phenotypes or continuous phenotypes</li>
		<li><em>gene sets database (file)</em>: same as used in step (1)</li>
		<li><em>top n</em>: set to a number equal to or less than the number of gene sets ssGSEAProjection identifies to be associated with the phenotype in the given&nbsp;<em>direction</em>&nbsp;(i.e., positive or negative).</li>
	</ol>
	</li>
</ol>

<p style="line-height: 20.8px;">For gene set collections that do not utilize _UP and _DN suffixes at the ends of set names, the&nbsp;<em>combine mode&nbsp;</em>parameter option is irrelevant as all the modes give the same output.</p>

<p style="line-height: 20.8px; margin-left: 0.5in;">As of MSigDB v5.0, released March 2015, four collections contain gene sets with _UP and _DN suffixes. These are&nbsp;<em>C2.all</em>,&nbsp;<em>C2.CGP</em>,&nbsp;<em>C6.all</em>, and&nbsp;<em>C7.all</em>. Some gene sets, including many in MSigDB, are derived from previous gene expression studies with up-regulated and down-regulated genes divided into separate gene sets denoted by an _UP or _DN suffix appended to otherwise identical set names. The ssGSEAProjection module can recombine these paired sets and calculate a combined enrichment score for each pair. See&nbsp;<a href="http://www.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection">ssGSEAProjection module documentation</a>&nbsp;for more information.</p>

<p style="line-height: 20.8px;">For gene set collections that utilize _UP and _DN suffixes, set ssGSEAProjection parameter&nbsp;<em>combine mode</em>&nbsp;to&nbsp;<em>combine.off</em>. Do not use the other two options,&nbsp;<em>combine.add&nbsp;</em>or&nbsp;<em>combine.replace</em>, as they create new combined gene sets that lack either suffix and which are not represented in the input gene set collection unless you will be providing ConstellationMap with a gene set collection that includes the new combined gene sets. ConstellationMap requires that the gene sets specified in the input GCT file match a subset of the gene sets provided.</p>

<ul style="line-height: 20.8px;">
	<li><em>Combine.replace</em>&nbsp;replaces the original paired gene sets with a new combined gene set and thus reduces the number of gene sets in the output by half the paired sets.</li>
	<li><em>Combine.add</em>&nbsp;mode keeps the original paired sets and independent scores and adds the new combined set with combined score to the output GCT file, increasing the number of gene sets by the number of paired sets.</li>
</ul>

<p><strong>Note:</strong>&nbsp;while the upstream GSEA, GSEAPreranked, and ssGSEAProjection modules all permit selection of multiple gene set files,&nbsp;<strong>this is not&nbsp;supported by ConstellationMap at this time.</strong>&nbsp; As a workaround, you can combine the gene set files by hand and upload the single combined file.</p>

<h2>Algorithm</h2>

<p style="line-height: 20.8px;">Constellation Map operates on the provided gene set enrichment scores, estimating the probability density functions of gene set and phenotypic class variables using kernel density estimation. These density functions are subsequently used to calculate normalized mutual information (NMI) scores for each gene set, which capture the association between each gene set’s enrichment scores and phenotypic classes (Eq. 3). Since NMI scores do not inherently contain information on direction of association, we apply the sign of the Pearson correlation (Eq. 4).</p>

<p style="line-height: 20.8px;">Next, the top N NMI-scoring gene sets are selected for projection onto the circular space. NMI scores are calculated pairwise across the N gene sets quantify the similarity between their enrichment profiles. These pairwise NMI scores are converted into dissimilarity scores,&nbsp;<em>d</em>&nbsp;= 1 −&nbsp;<em>NMI</em>, which has been mathematically proven to be a true distance metric. An&nbsp;<em>N</em>-by-<em>N</em>&nbsp;distance matrix&nbsp;<em>D</em>&nbsp;is constructed containing these pairwise distances&nbsp;<em>d</em>.</p>

<p style="line-height: 20.8px;">The circular Constellation Map plot is built using the multidimensional scaling projection R package SMACOF version 1.5–0. An angular distance matrix Δ is calculated by minimizing the objective function (Eq. 5), where&nbsp;<em>δ<sub>ij</sub></em>&nbsp;is the angular distance and&nbsp;<em>d<sub>ij</sub></em>&nbsp;is the original distance (stored in&nbsp;<em>D</em>) between gene sets&nbsp;<em>i</em>&nbsp;and&nbsp;<em>j</em>. The gene sets are plotted as points distributed about the origin. Angular distance between two gene sets is determined from Δ and is proportional to the similarity of the gene sets’ enrichment profiles. Radial distance (i.e., distance to the origin) indicates the gene set’s association with respect to the phenotype (1 –&nbsp;<em>NMI</em>).</p>

<p style="line-height: 20.8px;">To provide context of gene overlap between sets, pairwise Jaccard indices are calculated across the gene sets. (The Jaccard index is equal to the number of genes shared by two sets divided by the number of genes in their union.) For pairs with Jaccard indices greater than a given threshold, edges are drawn connecting the respective nodes; the thickness of each edge is proportional to the Jaccard index.</p>

<p style="line-height: 20.8px;"><img alt="" src="./content_eqtable.png" style="width: 600px; height: 263px;"></p>

<h2>References</h2>

<ol style="line-height: 20.8px;">
	<li>Barbie,D.A. et al. (2009) Systematic RNA interference reveals that oncogenic KRAS-driven cancers require TBK1. Nature, 462, 108–112.</li>
	<li>Dennis,G. et al. (2003) DAVID: Database for Annotation, Visualization, and Integrated Discovery. Genome Biol., 4, P3.</li>
	<li>Grossmann,S. et al. (2007) Improved detection of overrepresentation of Gene-Ontology annotations with parent child analysis. Bioinformatics, 23, 3024–3031.</li>
	<li>Jaccard,P. (1901) Étude comparative de la distribution florale dans une portion des Alpes et des Jura. Bulletin de la Société Vaudoise des Sciences Naturelles, 37, 547–579.</li>
	<li>Leeuw,J.d. and Meir,P. (2009) Multidimensional Scaling using Majorization: SMACOF in R. J. Statist. Software, 31, 1–30.</li>
	<li>Subramanian,A. et al. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proc. Natl. Acad. Sci. USA, 102, 15545–15550.</li>
	<li>Tan,Y. et al. (2014) Gene signatures related to B-cell proliferation predict influenza vaccine-induced antibody response. Eur. J. Immunol., 44, 285–295.</li>
	<li>Tan,Y. et al. (2015) Constellation Map: Downstream visualization and interpretation of gene set enrichment results. F1000Research, 4, 167.</li>
	<li>Vinh,N.X. et al. (2010) Information theoretic measures for clusterings comparison: variants, properties, normalization and correction for chance. J. Mach. Learn. Res., 11, 2837–2854.</li>
</ol>

	</div>
</div>
<div class="row">
	<div class="col-md-12">
		<h2>Parameters</h2>

<table class="table table-striped" style="line-height: 20.8px;">
	<thead>
		<tr>
			<th>Name</th>
			<th>Description</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>input gct file&nbsp;<span style="color: red;">*</span></td>
			<td>
			<p>This is a tab-delimited text file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GCT">GCT format</a>&nbsp;containing gene set enrichment data outputted from an ssGSEAProjection (single sample gene set enrichment analysis projection) module job.</p>
			ConstellationMap assumes that the input GCT file is a gene set enrichment file outputted by an ssGSEAProjection module job. Rows should correspond to gene sets while columns should correspond to sample names. The gene sets specified in the GCT file must be a subset of the gene sets listed in the accompanying&nbsp;<em>gene sets database&nbsp;</em>or<em>&nbsp;gene sets file</em>&nbsp;(see below).</td>
		</tr>
		<tr>
			<td>input cls file&nbsp;<span style="color: red;">*</span></td>
			<td>A space-delimited text file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#CLS">CLS format</a>&nbsp;containing two-phenotype categorical labels (e.g., tumor vs. normal) or continuous phenotype labels (e.g., time series). For each sample in the corresponding expression dataset the CLS file assigns a label or numerical value for the phenotype.</td>
		</tr>
		<tr>
			<td>gene sets database&nbsp;</td>
			<td>
			<p>This drop-down menu allows you to select gene sets from the&nbsp;<a href="http://www.msigdb.org/">Molecular Signatures Database (MSigDB)</a>&nbsp;on the GSEA website. This menu provides access to collections from MSigDB version 5.0.</p>

			<p>If you want to use files from an earlier version of MSigDB, you will need to download that file from the archived releases on the website and specify it in the gene sets file parameter.</p>

			<p>You can also upload a&nbsp;gene set file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMT">GMT</a>&nbsp;or&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMX">GMX</a>&nbsp;format. Provide your gene set file here if it is unavailable from the drop-down menu.</p>

			<p><strong>Note:</strong>&nbsp;while the upstream GSEA, GSEAPreranked, and ssGSEAProjection modules all permit selection of multiple gene set files,&nbsp;<strong>this is not supported by ConstellationMap at this time.</strong>&nbsp; As a workaround, you can combine the gene set files by hand and upload the single combined file.</p>
			</td>
		</tr>
		<tr>
			<td>gene sets file&nbsp;</td>
			<td>A gene set file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMT">GMT</a>&nbsp;or&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMX">GMX</a>&nbsp;format. Provide your gene set file here if it is unavailable from the drop-down menu under&nbsp;<em>gene sets database</em>.</td>
		</tr>
		<tr>
			<td>top n&nbsp;<span style="color: red;">*</span></td>
			<td>This is a positive integer indicating the number of top NMI-scoring gene sets to display in the final plot. This parameter must be greater than 2 but less than or equal to the total number of enriched gene sets.</td>
		</tr>
		<tr>
			<td>direction&nbsp;<span style="color: red;">*</span></td>
			<td>
			<p>This drop-down menu allows you to select the direction of correlation, “positive” or “negative”, in which ConstellationMap will investigate association to the&nbsp;<em>target class</em>.</p>

			<p>If “positive” is chosen then gene sets that are more positively associated with the&nbsp;<em>target class&nbsp;</em>will be placed closer to the center in the resulting radial plot. If “negative” is chosen then gene sets that are more negatively associated with the&nbsp;<em>target class&nbsp;</em>will be placed closer to the center in the resulting radial plot.</p>
			Default: “positive”</td>
		</tr>
		<tr>
			<td>image format&nbsp;<span style="color: red;">*</span></td>
			<td>
			<p>This drop-down menu allows you to select an image format for the outputted static plot and heat map, either PNG (raster graphics) or PDF (vector graphics).</p>
			Default: “PNG”</td>
		</tr>
		<tr>
			<td>jaccard threshold&nbsp;<span style="color: red;">*</span></td>
			<td>
			<p>This is a number between 0 and 1 indicating the Jaccard Index threshold above which connecting edges between gene sets will be drawn in the resulting plot.</p>

			<p>ConstellationMap measures the overlap between all gene set pairs using a Jaccard Index metric. The Jaccard Index is equal to the number of genes in the intersect of the two sets divided by the number of genes in their union.</p>

			<p>Edges are drawn between gene set pairs if their Jaccard Index is greater than the threshold parameter given here. This affects both the static and interactive plots.</p>
			Default: 0.1</td>
		</tr>
		<tr>
			<td>target class&nbsp;</td>
			<td>
			<p>This is a phenotype label, indicating the phenotypic category against which ConstellationMap will measure the association of various gene sets based on their enrichment. In the outputted visualization, target class will be plotted in the center of the radial plot.</p>
			This parameter must match one of the two phenotype labels specified in the second line of the&nbsp;<em>input cls file</em>. If this parameter is left blank, ConstellationMap will default to the first listed phenotype label in the&nbsp;<em>input cls file</em>.</td>
		</tr>
	</tbody>
</table>

<p style="line-height: 20.8px;"><span style="color: red;">*</span>&nbsp;- required</p>

	</div>
</div>
<div class="row">
	<div class="col-md-12">
		<h2>Input Files</h2>

<ol style="line-height: 20.8px;">
	<li>input gct file<br>
	Gene set enrichment data file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GCT">GCT format</a>.&nbsp;Each row of the GCT file matrix should represent a gene set while each column should represent a sample. Each matrix element should contain an enrichment score indicating the degree to which the given gene set’s members are coordinately up- or down-regulated in the given sample.&nbsp;Output from the&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/modules/docs/ssGSEAProjection/">ssGSEAProjection module</a>&nbsp;can be directly input into ConstellationMap.</li>
	<li>input cls file<br>
	Class designation file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#CLS">CLS format</a>.</li>
	<li>gene sets file or database<br>
	Gene sets file in&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMT">GMT</a>&nbsp;or&nbsp;<a href="https://www.broadinstitute.org/cancer/software/genepattern/file-formats-guide#GMX">GMX</a>&nbsp;format provided by the user or drawn from&nbsp;<a href="http://www.msigdb.org/">MSigDB</a>&nbsp;by selection from the pull-down menu.</li>
</ol>

<h2>Output Files</h2>

<ol style="line-height: 20.8px;">
	<li>Visualizer.html<br>
	<span style="line-height: 1.6;">This is an interactive in-browser plot powered by JavaScript. Open this link in a new tab to view the ConstellationMap radial plot. This visualizer allows users to quickly identify and annotate overlapping genes and interrogate gene set clusters. Metadata such as Jaccard indices and coordinates are quickly made available.</span>

	<p>When saving a ConstellationMap job to the GenePattern file system, users should ensure that Visualizer.html resides in the same folder as the two accompanying ODF files in order to continue using the visualizer.</p>
	This is the&nbsp;<strong>preferred</strong>&nbsp;method for visualizing ConstellationMap results.</li>
	<li>&lt;input.gct.file&gt;.&lt;gene.sets.file&gt;.&lt;target.class&gt;.HEATMAP.png&nbsp;<em>or</em>&nbsp;&lt;input.gct.file&gt;.&lt;gene.sets.file&gt;.&lt;target.class&gt;.HEATMAP.pdf<br>
	<span style="line-height: 1.6;">This image is a heat map of the top n NMI-scoring gene sets in either PNG or PDF format, depending on the chosen&nbsp;</span><em style="line-height: 1.6;">image format</em><span style="line-height: 1.6;">&nbsp;parameter.</span><br>
	Rows denote gene sets ranked by NMI score while columns denote samples. Cell colors from red (positive) to blue (negative) indicate ssGSEA enrichment scores. NMI, AUC, AUC p-values, t-test scores, and t-test score p-values for each gene set appear to the right. The horizontal yellow and purple bar at the top indicate the two different phenotypic classes.</li>
	<li>&lt;input.gct.file&gt;.&lt;gene.sets.file&gt;.&lt;target.class&gt;.REPORT.txt<br>
	This is a tab-delimited text file containing calculated statistical metrics for the&nbsp;<em>top n&nbsp;</em>NMI-scoring gene sets.</li>
</ol>

<ul style="line-height: 20.8px; margin-left: 40px;">
	<li>NMI: Normalized mutual information.</li>
	<li>AUC: Area under the curve.</li>
	<li>AUC.pval: Area under the curve p-value.</li>
	<li>t.stat: t-statistic.</li>
	<li>t.pval: t-statistic p-value.</li>
</ul>

<ol start="4" style="line-height: 20.8px;">
	<li>&lt;input.gct.file&gt;.&lt;gene.sets.file&gt;.&lt;target.class&gt;.CONSTELLATION_MAP.png&nbsp;<em>or&nbsp;</em>&lt;input.gct.file&gt;.&lt;gene.sets.file&gt;.&lt;target.class&gt;.CONSTELLATION_MAP.pdf<br>
	This is a static image of the ConstellationMap radial plot in either PNG or PDF format, depending on the chosen&nbsp;<em>image format&nbsp;</em>parameter.</li>
	<li>ConstellationMap.plot.data.nodes.odf<br>
	This is a tab-delimited text file in ODF format that contains data on node coordinates as well as metadata (e.g., gene set membership, filenames) for plotting by&nbsp;<em>Visualizer.html</em>.</li>
	<li>ConstellationMap.plot.data.edges.odf<br>
	This is a tab-delimited text file in ODF format that contains data on edge locations as well as metadata (e.g., Jaccard Indices, filenames) for plotting by&nbsp;<em>Visualizer.html</em>.</li>
</ol>

<h2>Requirements</h2>

<h3>Supported Browsers</h3>

<table border="1" cellpadding="1" cellspacing="1" style="line-height: 20.8px; width: 250px;">
	<tbody>
		<tr>
			<td><strong>Mozilla Firefox:</strong></td>
			<td>v4–46</td>
		</tr>
		<tr>
			<td><strong>Google Chrome:</strong></td>
			<td>v13–41</td>
		</tr>
		<tr>
			<td><strong>Microsoft IE:</strong></td>
			<td>v10–11</td>
		</tr>
	</tbody>
</table>

<p style="line-height: 20.8px;"><em>The module developers highly suggest using the latest version of Firefox or Chrome. JavaScript must be enabled.</em></p>

<h3>Platform Dependencies</h3>

<table border="1" cellpadding="1" cellspacing="1" style="line-height: 20.8px; width: 250px;">
	<tbody>
		<tr>
			<td><strong>Task Type:</strong></td>
			<td>Statistical Methods</td>
		</tr>
		<tr>
			<td><strong>CPU Type:</strong></td>
			<td>any</td>
		</tr>
		<tr>
			<td><strong>OS:</strong></td>
			<td>any</td>
		</tr>
		<tr>
			<td><strong>Language:</strong></td>
			<td>R (v3.0)</td>
		</tr>
	</tbody>
</table>

	</div>
</div>

<h2>Platform Dependencies</h2>

<div class="row">
	<div class="col-md-3">
		<p><strong>Task Type:</strong><br>
		Statistical Methods</p>
	</div>
	<div class="col-md-3">
		<p><strong>CPU Type:</strong><br>
		any</p>
	</div>
	<div class="col-md-3">
		<p><strong>Operating System:</strong><br>
		any</p>
	</div>
	<div class="col-md-3">
		<p><strong>Language:</strong><br>
		R</p>
	</div>
</div>

<h2>Version Comments</h2>
<table class="table table-striped">
	<thead>
		<tr>
			<th>Version</th>
			<th>Release Date</th>
			<th>Description</th>
		</tr>
	</thead>
	<tbody>
			<tr>
				<td>1.4.16</td>
				<td>2021-4-22</td>
				<td>Fixed minor typo.</td>
			</tr>
			<tr>
				<td>1.4.15</td>
				<td>2021-4-9</td>
				<td>More flexible CLS parsing</td>
			</tr>
			<tr>
				<td>1.4.14</td>
				<td>2021-4-2</td>
				<td>Updated for MSigDB v7.4</td>
			</tr>
			<tr>
				<td>1.4.13</td>
				<td>2021-3-22</td>
				<td>Updated for MSigDB v7.3</td>
			</tr>
			<tr>
				<td>1.4.12</td>
				<td>2020-9-23</td>
				<td>Updated for MSigDB v7.2</td>
			</tr>
			<tr>
				<td>1.4.11</td>
				<td>2020-4-2</td>
				<td>Updated for MSigDB v7.1</td>
			</tr>
			<tr>
				<td>1.4.10</td>
				<td>2019-10-16</td>
				<td>Updated for MSigDB v7.0</td>
			</tr>
			<tr>
				<td>1.4.9</td>
				<td>2018-07-17</td>
				<td>Beta release for the MSigDB v6.2 update</td>
			</tr>
	</tbody>
</table>

	</div>
</div>
<div class="footer fluid">
	<div id="footer" class="container">
		<div class="row">
    	<div class="col-md-12">
				<p class="copy-right-block"><a href="http://www.broadinstitute.org/"><img alt="Broad logo" class="footer-logo" src="./broad-logo.png"></a>©2021 <a href="http://www.broadinstitute.org/">Broad Institute of MIT &amp; Harvard</a></p>
			</div>
		</div>
	</div>
</div>

</body></html>